packages/noya-designsystem/src/components/InputField.tsx

```diff
@@ -64,35 +112,62 @@ function InputFieldLabel({ children = false }: InputFieldLabelProps) {
 const InputElement = styled(TextInput)<{
   labelPosition: LabelPosition;
   labelSize: number;
-}>(({ theme, labelPosition, labelSize }) => ({
-  ...theme.textStyles.small,
-  color: theme.colors.text,
-  width: '0px', // Reset intrinsic width
-  flex: '1 1 0px',
-  position: 'relative',
-  border: '0',
-  outline: 'none',
-  minWidth: '0',
-  textAlign: 'left',
-  alignSelf: 'stretch',
-  borderRadius: '4px',
-  paddingTop: '4px',
-  paddingBottom: '4px',
-  paddingLeft: labelPosition === 'start' ? `${6 + labelSize + 6}px` : '6px',
-  paddingRight: labelPosition === 'start' ? '6px' : `${6 + labelSize + 6}px`,
-  background: theme.colors.inputBackground,
-  '&:focus': {
-    boxShadow: `0 0 0 2px ${theme.colors.primary}`,
-  },
-}));
+  hasLabel: boolean;
+  hasDropdown: boolean;
+  textAlign?: Property.TextAlign;
+  disabled?: boolean;
+}>(
+  ({
+    theme,
+    labelPosition,
+    labelSize,
+    hasDropdown,
+    textAlign,
+    disabled,
+    hasLabel,
+  }) => ({
+    ...theme.textStyles.small,
+    color: disabled ? theme.colors.textDisabled : theme.colors.text,
+    width: '0px', // Reset intrinsic width
+    flex: '1 1 0px',
+    position: 'relative',
+    border: '0',
+    outline: 'none',
+    minWidth: '0',
+    textAlign: textAlign ?? 'left',
+    alignSelf: 'stretch',
+    borderRadius: '4px',
+    paddingTop: '4px',
+    paddingBottom: '4px',
+    paddingLeft:
+      labelPosition === 'start'
```

Comment:
If we do `hasLabel ?` first, we can probably simplify these a little

```ts
paddingLeft: hasLabel && labelPosition === 'start' ? `${6 + labelSize + 6}px` : '6px'
```

Maybe paddingRight can be simpler too